src/pages/settings/Wallet/ActivatePhysicalCardPage.tsx

```diff
@@ -55,30 +54,30 @@ function ActivatePhysicalCardPage({
     const [lastFourDigits, setLastFourDigits] = useState('');
     const [lastPressedDigit, setLastPressedDigit] = useState('');
 
-    const domainCards = CardUtils.getDomainCards(cardList)[domain] ?? [];
-    const physicalCard = domainCards.find((card) => !card.nameValuePairs?.isVirtual);
-    const cardID = physicalCard?.cardID ?? 0;
-    const cardError = ErrorUtils.getLatestErrorMessage(physicalCard ?? {});
+    const inactiveCard = cardList?.[cardID];
+    const cardError = ErrorUtils.getLatestErrorMessage(inactiveCard ?? {});
 
     const activateCardCodeInputRef = useRef<MagicCodeInputHandle>(null);
 
     /**
      * If state of the card is CONST.EXPENSIFY_CARD.STATE.OPEN, navigate to card details screen.
      */
     useEffect(() => {
-        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-        if (physicalCard?.isLoading || cardList?.[cardID]?.state !== CONST.EXPENSIFY_CARD.STATE.OPEN) {
+        if (inactiveCard?.state !== CONST.EXPENSIFY_CARD.STATE.OPEN || inactiveCard?.isLoading) {
             return;
         }
 
-        Navigation.navigate(ROUTES.SETTINGS_WALLET_DOMAINCARD.getRoute(domain));
-    }, [cardID, cardList, domain, physicalCard?.isLoading]);
+        Navigation.navigate(ROUTES.SETTINGS_WALLET_DOMAINCARD.getRoute(cardID));
+    }, [cardID, cardList, inactiveCard?.isLoading, inactiveCard?.state]);
 
     useEffect(
         () => () => {
-            CardSettings.clearCardListErrors(cardID);
+            if (!inactiveCard?.cardID) {
+                return;
+            }
+            CardSettings.clearCardListErrors(inactiveCard?.cardID);
```

Comment:
Error is not cleared when page is reloaded, https://github.com/Expensify/App/issues/51396